src/main/scala/fi/oph/kouta/repository/sessionDAO.scala

```diff
@@ -0,0 +1,95 @@
+package fi.oph.kouta.repository
+
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+import fi.oph.kouta.security.{CasSession, Role, ServiceTicket, Session}
+import slick.dbio.DBIO
+import slick.jdbc.PostgresProfile.api._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+
+trait SessionDAO {
+  def delete(ticket: ServiceTicket): Unit
+
+  def delete(id: UUID): Unit
+
+  def store(session: Session): UUID
+
+  def get(id: UUID): Option[Session]
+}
+
+object SessionDAO extends SessionDAO with SessionSQL {
+  import KoutaDatabase.runBlocking
+
+  override def store(session: Session): UUID = session match {
+    case CasSession(ServiceTicket(ticket), personOid, roles) =>
+      val id = UUID.randomUUID()
+      runBlocking(storeCasSession(id, ticket, personOid, roles), timeout = Duration(1, TimeUnit.MINUTES))
+      id
+  }
+
+  def store(session: CasSession, id: UUID) =
+    runBlocking(storeCasSession(id, session.casTicket.s, session.personOid, session.roles), timeout = Duration(1, TimeUnit.MINUTES))
+
+
+  override def delete(id: UUID): Unit = {
+    runBlocking(deleteSession(id), timeout = Duration(10, TimeUnit.SECONDS))
+  }
+
+  override def delete(ticket: ServiceTicket): Unit = {
+    runBlocking(deleteSession(ticket), timeout = Duration(10, TimeUnit.SECONDS))
+  }
+
+  override def get(id: UUID): Option[Session] = {
+    runBlocking(getSession(id), timeout = Duration(2, TimeUnit.SECONDS)).map {
+      case (casTicket, personOid) =>
+        val roolit = runBlocking(searchRoolitBySessio(id), Duration(2, TimeUnit.SECONDS))
+        CasSession(ServiceTicket(casTicket.get), personOid, roolit.map(Role(_)).toSet)
+    }
+  }
+
+}
+
+sealed trait SessionSQL extends SQLHelpers {
+
+  protected def storeCasSession(
+      id: UUID,
```

Comment:
https://docs.scala-lang.org/style/types.html

> All public methods should have explicit type annotations. Type inference may break encapsulation in these cases, because it depends on internal method and class details. Without an explicit type, a change to the internals of a method or val could alter the public API of the class without warning, potentially breaking client code. Explicit type annotations can also help to improve compile times.

Reply:
Tiedän kyllä tämän ja olen rikkonut tätä sääntöä tässä kohdassa tietoisesti! En tarvitse sinulta opetusta, kiitos vain!